app/main.py

```diff
@@ -1,6 +1,24 @@
 from typing import Callable
+import functools
+cache_dict = {}
```

Comment:
What if you're gonna wrap several functions with the decorator, will it work? You will have same cache for all wrapped functions. You should create this variable inside of `cache` function

Reply:
But why, if I create "cache_dict" inside of decorator cache isn't it gonna re-assining itself to empty list? I done that to create inside this "cache_dict" a key with name of a decorated function and a value as another dict with function own cache. @LiudmylaKulinchenko

Reply:
@dartomOOv no, decorator is initialized only once(if you use syntax sugar), so the code out of wrapper is run only once. But wrapper is called each time you call the function that is wrapper by this decorator. You can set breakpoint on cache variable inside decorator, call **decorated** function and check how many breakpoints you'll get
Example:
```
from typing import Callable


def cache(func: Callable) -> Callable:
    cache_dict = {}

    def wrapper(*args) -> int:
        nonlocal cache_dict

        if args not in cache_dict:
            cache_dict[args] = func(*args)
            print("Calculating new result")
        else:
            print("Getting from cache")

        return cache_dict[args]

    return wrapper

@cache
def func(arg):
    return arg + 1

func(1)
func(2)
func(2)
# cache_dict = {} will be called only once
```
It's because you use syntax sugar and when python runs the code, it knows that **cache** is decorator.
However if you remove decorator from **func** and run it in another way(without syntax sugar) like this:
```
cache(func)(1)
cache(func)(2)
cache(func)(2)
```
then yes, cache_dict will be cleared on each call. This is the magic of syntax sugar :)

Reply:
Ok, I see now. Thx very much!